apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ResponseAdapterCache.kt

```diff
@@ -1,12 +1,22 @@
 package com.apollographql.apollo.api
 
+import com.apollographql.apollo.api.internal.ResponseAdapter
+import com.apollographql.apollo.api.internal.json.JsonReader
+import com.apollographql.apollo.api.internal.json.JsonWriter
+import com.apollographql.apollo.api.internal.json.Utils.readRecursively
+import com.apollographql.apollo.api.internal.json.Utils.writeToJson
+import kotlin.jvm.Synchronized
+
 /**
  * A wrapper around a Map of [CustomScalarAdapter] that allows to easily retrieve an adapter for the given [CustomScalar]
  */
-class CustomScalarAdapters(val customScalarAdapters: Map<CustomScalar, CustomScalarAdapter<*>>) {
+class ResponseAdapterCache(val customScalarAdapters: Map<CustomScalar, CustomScalarAdapter<*>>) {
 
   private val adapterByGraphQLName = customScalarAdapters.mapKeys { it.key.graphqlName }
 
+  private val adapterByQueryName = ThreadSafeMap<String, ResponseAdapter<*>>()
+  private val adapterByFragmentName = ThreadSafeMap<String, ResponseAdapter<*>>()
```

Comment:
so if accessor APIs for this class is `synchronized` then we don't need `ThreadSafeMap`, right?

Reply:
Gaaa, sorry I forgot to remove the `@Synchronized` from `ResponseAdatperCache`. We need `ThreadSafeMap` for native